tensornetwork/tn_keras/test_layer.py

```diff
@@ -104,3 +120,84 @@ def test_num_parameters(dummy_data):
       decomp_size * output_dim) + output_dim
 
   np.testing.assert_equal(expected_num_parameters, model.count_params())
+
+
+def test_mpo_num_parameters(dummy_data):
+  # Disable the redefined-outer-name violation in this function
+  # pylint: disable=redefined-outer-name
+  data, _ = dummy_data
+  input_dim = data.shape[1]
+  output_dim = data.shape[1]
+  num_nodes = int(math.log(data.shape[1], 4))
+  bond_dim = 8
+
+  model = Sequential()
+  model.add(
+      DenseMPO(output_dim,
+               num_nodes=num_nodes,
+               bond_dim=bond_dim,
+               use_bias=True,
+               activation='relu',
+               input_dim=input_dim))
+
+  in_leg_dim = int(input_dim**(1. / num_nodes))
+  out_leg_dim = int(output_dim**(1. / num_nodes))
+
+  # num_params = num_edge_node_params + num_middle_node_params + bias_params
+  expected_num_parameters = (2 * in_leg_dim * bond_dim * out_leg_dim) + (
+      (num_nodes - 2) * in_leg_dim * bond_dim * bond_dim *
+      out_leg_dim) + output_dim
+
+  np.testing.assert_equal(expected_num_parameters, model.count_params())
+
+
+def test_config(make_model):
+  # Disable the redefined-outer-name violation in this function
+  # pylint: disable=redefined-outer-name
+  model = make_model
+
+  expected_num_parameters = model.layers[0].count_params()
+
+  # Serialize model and use config to create new layer
+  model_config = model.get_config()
+  layer_config = model_config['layers'][0]['config']
+  if 'mpo' in model.layers[0].name:
+    new_model = DenseMPO.from_config(layer_config)
+  elif 'decomp' in model.layers[0].name:
+    new_model = DenseDecomp.from_config(layer_config)
+
+  # Build the layer so we can count params below
+  new_model.build(layer_config['batch_input_shape'])
+
+  # Check that original layer had same num params as layer built from config
+  np.testing.assert_equal(expected_num_parameters, new_model.count_params())
+
+
+def test_model_save(dummy_data, make_model):
+  # Disable the redefined-outer-name violation in this function
+  # pylint: disable=redefined-outer-name
+
+  data, labels = dummy_data
+  model = make_model
+  model.compile(optimizer='adam',
+                loss='binary_crossentropy',
+                metrics=['accuracy'])
+
+  # Train the model for 5 epochs
+  model.fit(data, labels, epochs=5, batch_size=32)
+
+  # Save model to a h5 file, then load from file
+  model.save('test_model.h5')
```

Comment:
You might also want to test saving in format=tf (which is SavedModel).

Reply:
Sounds good, added a test for the SavedModel format.